src/plugins/histogram/index.js

```diff
@@ -0,0 +1,215 @@
+import videojs from 'video.js';
+
+import './histogram.scss';
+
+// Default options for the plugin.
+let defaults = {};
+
+/**
+ * Function to invoke when the player is ready.
+ *
+ * @function onPlayerReady
+ * @param    {Player} player
+ *           A Video.js player object.
+ *
+ * @param    {Object} [options={}]
+ *           A plain object containing options for the plugin.
+ */
+const onPlayerReady = function onPlayerReady(player, options) {
+  player.addClass('vjs-histogram');
+  player.histogram = new HistogramPlugin(player, options);
+};
+
+/**
+ * A video.js plugin.
+ *
+ * In the plugin function, the value of `this` is a video.js `Player`
+ * instance. You cannot rely on the player being in a "ready" state here,
+ * depending on how the plugin is invoked. This may or may not be important
+ * to you; if not, remove the wait for "ready"!
+ *
+ * @function histogram
+ * @param    {Object} [options={}]
+ *           An object of options left to the plugin author to define.
+ */
+const histogram = function histogram(options) {
+  let _this = this;
+
+  this.ready(function () {
+    onPlayerReady(_this, videojs.mergeOptions(defaults, options));
+  });
+};
+
+/**
+ * Histogram class.
+ *
+ * This class performs all functions related to displaying the Histogram bar.
+ */
+const HistogramPlugin = (function () {
```

Comment:
For next time i recommend writing this using Es6 class syntax

Reply:
This is building on top of the seek-thumbnails "boilerplate", so uses the same structure, we can refactor, but this was quite helpful and easy to use